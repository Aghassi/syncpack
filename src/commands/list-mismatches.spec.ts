import * as mock from '../../test/mock';

describe('listMismatches', () => {
  let listMismatches: any;
  let log: any;

  afterEach(() => {
    jest.restoreAllMocks();
  });

  beforeEach(() => {
    jest.mock('./lib/log', () => ({ log: jest.fn() }));
    listMismatches = require('./list-mismatches').listMismatches;
    log = require('./lib/log').log;
  });

  it('outputs all dependencies installed with different versions', () => {
    const wrappers = [mock.wrapper('a', ['foo@0.1.0']), mock.wrapper('b', ['foo@0.2.0'])];
    listMismatches(['dependencies'], /./, wrappers);
    expect(log.mock.calls).toMatchSnapshot();
  });
});
