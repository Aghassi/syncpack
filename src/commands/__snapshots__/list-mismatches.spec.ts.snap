// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`listMismatches outputs all dependencies installed with different versions 1`] = `
Array [
  Array [
    "[31mâœ• foo[39m",
  ],
  Array [
    "[2m-[22m 0.1.0 [2min dependencies of undefined[22m",
  ],
  Array [
    "[2m-[22m 0.2.0 [2min dependencies of undefined[22m",
  ],
]
`;
